<launch>
    <arg name="set_base_frame" default="base_footprint"/>
    <arg name="set_odom_frame" default="odom"/>
    <arg name="set_map_frame"  default="map"/>
    <arg name="is_sim"  default="False"/>

    <!-- Laser -->
    <!-- <include file="$(find nav_main)/launch/lidar.launch">
        <param name="is_sim" value="$(arg is_sim)"/>
    </include> -->

    <include file="$(find robot_description)/launch/display.launch" />

    <node name="teleop_twist_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py"/>

    <!-- Gmapping -->
    <node pkg="gmapping" type="slam_gmapping" name="robot_slam_gmapping" output="screen">
        <param name="scan" value="scan"/>
        <param name="base_frame" value="$(arg set_base_frame)"/>
        <param name="odom_frame" value="$(arg set_odom_frame)"/>
        <param name="map_frame"  value="$(arg set_map_frame)"/>
        <remap from="scan" to="scan_filtered"/>
        <rosparam command="load" file="$(find nav_main)/config/gmapping_params.yaml" />
    </node>

    <!-- Save Map CMD. -->
    <!-- rosrun map_server map_saver static_map:=dynamic_map -->
</launch>